### DISCUSSION TITLE
Welcome to fluidpatcher Discussions!

### DISCUSSION THREAD

**Discussion by albedozero** [GeekFunkLabs/fluidpatcher#1] [General]
## 👋 Welcome!
  Discussions is a place to connect with other members of the fluidpatcher community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas for modifications/features.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

To get started, comment below with an introduction of yourself, a little about what you do, and your interest in/plans for using fluidpatcher.

**Comment by Tauzero** [comment]
I was so interested that I built one. I'm a programmer and a bassist, and wanted to use bass pedals with a flexible sound module, and this seemed like just the thing. I had a bit of a false start using the original build, and decided to simplify the hardware a little by using an I2C interface on the display. I also went for a 20x4 display rather than 16x2. I amended utils/stompbox.py and squishbox.py to use rows and columns defined in utils/hw_overlay.py rather than hard-coded values, and also amended utils/stompbox.py to use either I2C display or 16-wire display. I also changed the menus etc to take advantage of the larger display if present. If you'd like copies of the files, you're welcome to them.

**Reply by albedozero** [reply]
Sure! Had plans to build some more options into the code so it would be a help. Feel free to send files, or do a fork and pull request if you want to get fancy :). Would love to see pictures of your build as well!

**Reply by Tauzero** [reply]
Files attached - they're edited from 3.3, not sure if you've gone to 3.4 yet. ".txt" extension added to bypass Git's restrictions. I'm not confident enough to do a fork!

[hw_overlay.py.txt](https://github.com/albedozero/fluidpatcher/files/6346609/hw_overlay.py.txt)
[squishbox_i2c.py.txt](https://github.com/albedozero/fluidpatcher/files/6346610/squishbox_i2c.py.txt)
[stompboxpi_i2c.py.txt](https://github.com/albedozero/fluidpatcher/files/6346611/stompboxpi_i2c.py.txt)

**Reply by Tauzero** [reply]
And some photos too

![1_display_and_power](https://user-images.githubusercontent.com/63983755/115464245-b4286280-a224-11eb-9d5f-55880b896cfd.jpg)
![2_protoboard_bottom](https://user-images.githubusercontent.com/63983755/115464248-b4c0f900-a224-11eb-9599-5a7c520b95fd.jpg)
![3_protoboard_top](https://user-images.githubusercontent.com/63983755/115464250-b5598f80-a224-11eb-873a-780601b6c7e7.jpg)
![4_pi_in_lid](https://user-images.githubusercontent.com/63983755/115464252-b5598f80-a224-11eb-8846-8b55c33ebd6c.jpg)
![5_lid_fixings](https://user-images.githubusercontent.com/63983755/115464254-b5f22600-a224-11eb-8d12-73e91e15e2bf.jpg)
![6_internals](https://user-images.githubusercontent.com/63983755/115464257-b5f22600-a224-11eb-89de-ffca65cc01e1.jpg)
![6_pi_plus_protoboard](https://user-images.githubusercontent.com/63983755/115464259-b5f22600-a224-11eb-8ff4-4293a225ff9c.jpg)
![7_complete](https://user-images.githubusercontent.com/63983755/115464264-b68abc80-a224-11eb-942e-1e133c035a49.jpg)

**Reply by albedozero** [reply]
Nice! Is that a little 5V regulator on the power jack? If so, what is it? I've been looking for one of those. Your build looks nice and solid. Clean cuts too - is that careful filing or just skill? What tool do you use?

**Reply by Tauzero** [reply]
The power jack has a bridge rectifier to prevent embarrassing moments of reversed polarity, then a 5V regulator. The one I used was this one: https://www.ebay.co.uk/itm/191944441800 which can be used either as adjustable voltage or, by cutting a track and soldering across a couple of pads, as a fixed voltage. I cut the holes using a pillar drill to cut 12mm holes at each corner and a series of smaller holes almost linking them, then a jigsaw to cut the rough outline, then a power file to get the hole to size, and hand filing just to get the corners to right-angles. I've done a bit of a write-up on the whole thing:

The intention was to build a unit that could be powered by a guitar effects 9V power supply of sufficient current capacity, and outputting via a 5102 sound board to 1/4" jacks.

The 5.5/2.1mm power socket goes to a KBP307G bridge rectifier to provide reverse voltage protection (any bridge rectifier with a maximum current of 3A or more will do) and then to a buck converter set to 5V.

This 5V feed goes to a protoboard that is also used to mount the 5102 board and male headers for the display and footswitches. The 5102 board itself has a further male header for the output sockets.

With the way I've mounted this, the whole will fit into a 1590BBS Hammond box. I've made another one in a 1590C box which gives somewhat more room.

The display is mounted on brass M3 spacers with 6mm of stand-off which are epoxied into the box (Araldite Rapid was useless, J-B Weld has proved rather better). The Pi is mounted on 25mm M2.5 countersunk bolts with a nut and washer holding the Pi, then a 10mm nylon spacer, then the protoboard, then nylon M2.5 nuts. I've put heatshrink around the power supply components.

Note that the power connector must have the sleeve isolated from the case.

After final assembly, I found I was getting some audible interference. After disassembling it and experimenting with various things, I discovered it was caused when the metalwork around the Pi sockets touches the case. I'm going to file the hole a little larger and put heatshrink round each of the sockets.

**Comment by Tauzero** [comment]
As the Pi has a sound card and storage on it, and being able to play sound files may come in handy for gigs, I wrote a simple player which uses stompbox routines to decode button presses, and then, more ambitiously, integrated it with squishbox in something called (for want of a better name) squishsak, for Swiss Army knife. Some other changes to squishbox and stompbox to take advantage of using a 20x4 display too.

[squishbox_i2c.py.txt](https://github.com/albedozero/fluidpatcher/files/6385628/squishbox_i2c.py.txt)
[squishplayer.py.txt](https://github.com/albedozero/fluidpatcher/files/6385630/squishplayer.py.txt)
[squishsak.py.txt](https://github.com/albedozero/fluidpatcher/files/6385631/squishsak.py.txt)
[stompboxpi_i2c.py.txt](https://github.com/albedozero/fluidpatcher/files/6385632/stompboxpi_i2c.py.txt)

**Reply by albedozero** [reply]
Looks great! I was thinking sample player at first, but this is perfect for set break music! I was hoping someone would come up with some other apps for the stombpox code - I'll work on integrating these into the repository. SquishSAK - have you figured out my naming scheme? X-D

**Reply by Tauzero** [reply]
I've added in copying in files from USB too, into both the player and SquishSAK (or whatever name it gets, I'm low on inspiration!). I have set up the GitHub desktop so I can do a fork at some point. I think the player could be expanded to include playing a list as well as individual songs - it's one way of learning Python.
[squishplayer.py.txt](https://github.com/albedozero/fluidpatcher/files/6394435/squishplayer.py.txt)
[squishsak.py.txt](https://github.com/albedozero/fluidpatcher/files/6394437/squishsak.py.txt)

**Reply by Tauzero** [reply]
Haven't updated this for a while. I found there was an issue with squishsak - it worked OK when run standalone, but when I tried using it as a service, it failed to start correctly. As there weren't any gigs on the horizon, I didn't do anything more, but I shall resume playing with it shortly.
